packages/squiggle-lang/src/rescript/ReducerProject/ReducerProject.res

```diff
@@ -0,0 +1,396 @@
+let getIncludes = (this: t, sourceId: string): ProjectItem.T.includesType =>
+  this->T.Private.castToInternalProject->Private.getIncludes(sourceId)
+
+let getContinues = (this: t, sourceId: string): array<string> =>
+  this->T.Private.castToInternalProject->Private.getContinues(sourceId)
+
+// setContinues acts like an include hidden in the source. It is used to define a continuation.
+let setContinues = (this: t, sourceId: string, continues: array<string>): unit =>
+  this->T.Private.castToInternalProject->Private.setContinues(sourceId, continues)
+
+// This source is not continuing any other source. It is a standalone source.
+// Touches this source also.
+let removeContinues = (this: t, sourceId: string): unit =>
+  this->T.Private.castToInternalProject->Private.removeContinues(sourceId)
+
+// Gets includes and continues for a given source id. SourceId is depending on them
+let getDependencies = (this: t, sourceId: string): array<string> =>
+  this->T.Private.castToInternalProject->Private.getDependencies(sourceId)
+
+// Get source ids depending on a given source id.
+let getDependents = (this: t, sourceId: string): array<string> =>
+  this->T.Private.castToInternalProject->Private.getDependents(sourceId)
+
+// Get run order for all sources. It is a topological sort of the dependency graph.
+let getRunOrder = (this: t) => this->T.Private.castToInternalProject->Private.getRunOrder
+
+// Get run order for a given source id. It is a topological sort of the dependency graph.
+let getRunOrderFor = (this: t, sourceId: string) =>
+  this->T.Private.castToInternalProject->Private.getRunOrderFor(sourceId)
+
+// Parse includes so that you can load them before running. Use getIncludes to get the includes.
+// It is your responsibility to load the includes before running.
+let parseIncludes = (this: t, sourceId: string): unit =>
+  this->T.Private.castToInternalProject->Private.parseIncludes(sourceId)
+
+// Parse the source code if it is not done already. Use getRawParse to get the parse tree
+let rawParse = (this: t, sourceId: string): unit =>
+  this->T.Private.castToInternalProject->Private.rawParse(sourceId)
+
+// Runs the source code.
+// The code is parsed if it is not already done.
+// If it continues/includes another source then it will run that source also if is not already done.
+let run = (this: t, sourceId: string): unit =>
+  this->T.Private.castToInternalProject->Private.run(sourceId)
+
+// Runs all the sources.
+let runAll = (this: t): unit => this->T.Private.castToInternalProject->Private.runAll
+
+// WARNING" getExternalBindings will be deprecated. Cyclic directed graph problems
+// Get the bindings after running the source code.
+let getExternalBindings = (this: t, sourceId: string): ExternalExpressionValue.record =>
+  this->T.Private.castToInternalProject->Private.getExternalBindings(sourceId)
+
+//WARNING: externalResult will be deprecated. Cyclic directed graph problems
+let getExternalResult = (this: t, sourceId: string): option<
```

Comment:
There's a whole bunch of different functions here. I like that they're neat, but might like it more if they were cleaned up into a few Modules with different purposes. I imagine there are a few very distinct use cases where one might want to use these functions; it would be good to keep these separate.